app/controllers/network_router_controller.rb

```diff
@@ -462,14 +402,16 @@ def form_external_gateway(params)
         subnet = find_record_with_rbac(CloudSubnet, params[:cloud_subnet_id])
         options[:external_gateway_info][:external_fixed_ips] = [{:subnet_id => subnet.ems_ref}]
       end
-      options[:external_gateway_info][:enable_snat] = switch_to_bool(params[:enable_snat])
+      if params[:extra_attributes][:external_gateway_info][:enable_snat]
```

Comment:
Please use `params.fetch_path(:extra_attributes, :external_gateway_info, :enable_snat)` just in case `params` don't have `:extra_attributes` or `:external_gateway_info`.

Reply:
Done, thanks.